src/bun.js/bindings/webcore/JSEventTarget.cpp

```diff
@@ -223,6 +224,23 @@ static inline JSC::EncodedJSValue jsEventTargetPrototypeFunction_addEventListene
     EnsureStillAliveScope argument2 = callFrame->argument(2);
     auto options = argument2.value().isUndefined() ? false : convert<IDLUnion<IDLDictionary<AddEventListenerOptions>, IDLBoolean>>(*lexicalGlobalObject, argument2.value());
     RETURN_IF_EXCEPTION(throwScope, {});
+    // Emit a warning if listener is null, as it has no effect
+    if (!listener) {
+        auto& vm = JSC::getVM(lexicalGlobalObject);
+        String warningMessage;
+        if (argument1.value().isNull()) {
+            warningMessage = makeString("addEventListener called with null listener, which has no effect."_s);
```

Comment:
makeString is an unnecessary copy.
```suggestion
            warningMessage = "addEventListener called with null listener, which has no effect."_s;
```